keyboards/pica40/keymaps/default/keymap.c

```diff
@@ -0,0 +1,212 @@
+        XXXXXXX, KC_LGUI, KC_LALT, KC_LCTL, KC_LSFT, XXXXXXX,             LALT(KC_UP),   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_INS,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,             LALT(KC_DOWN), KC_HOME, KC_END,  KC_APP,  XXXXXXX, XXXXXXX,
+                                       XXXXXXX, XXXXXXX, XXXXXXX,     XXXXXXX, KC_TRNS, XXXXXXX
+    ),
+
+    [_NUMBER] = LAYOUT(
+                 KC_BSLS, KC_7,    KC_8,    KC_9,    KC_0,                XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        KC_LCTL, KC_COMM, KC_4,    KC_5,    KC_6,    KC_LBRC,             XXXXXXX, KC_RSFT, KC_RCTL, KC_LALT, KC_RGUI, XXXXXXX,
+        KC_DEL,  KC_DOT,  KC_1,    KC_2,    KC_3,    KC_RBRC,             XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+                                      KC_SPC, KC_TRNS, XXXXXXX,       XXXXXXX, XXXXXXX, XXXXXXX
+    ),
+
+    [_SYMBOL] = LAYOUT(
+                 LSFT(KC_BSLS), LSFT(KC_7), LSFT(KC_8), LSFT(KC_9), LSFT(KC_0),               XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        KC_LCTL, LSFT(KC_COMM), LSFT(KC_4), LSFT(KC_5), LSFT(KC_6), LSFT(KC_LBRC),            XXXXXXX, KC_RSFT, KC_RCTL, KC_LALT, KC_RGUI, XXXXXXX,
+        KC_BSPC, LSFT(KC_DOT),  LSFT(KC_1), LSFT(KC_2), LSFT(KC_3), LSFT(KC_RBRC),            XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+                                                    LSFT(KC_GRV), TG(_GAME), KC_TRNS,     XXXXXXX, XXXXXXX, XXXXXXX
+    ),
+
+    [_FUNC] = LAYOUT(
+                 KC_F12, KC_F7,   KC_F8,   KC_F9,   KC_PSCR,            XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        KC_LCTL, KC_F11, KC_F4,   KC_F5,   KC_F6,   KC_PAUS,            XXXXXXX, KC_RSFT, KC_RCTL, KC_LALT, KC_RGUI, XXXXXXX,
+        KC_ENT,  KC_F10, KC_F1,   KC_F2,   KC_F3,   KC_CAPS,            XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+                                     KC_TRNS, KC_MNXT, KC_MPLY,     KC_TRNS, XXXXXXX, XXXXXXX
+    )
+};
+
+bool get_tapping_force_hold(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        // allow multiple space, backspace, delete
+        case SYM_SPC:
+        case NAV_BSPC:
+        case RALT_DEL:
+            return false;
+        default:
+            return true;
+    }
+}
+
+uint16_t get_tapping_term(uint16_t keycode, keyrecord_t *record) {
+    // different tapping term for different fingers
+    switch (keycode) {
+        // pinkies
+        case HOME_A:
+        case HOME_SCLN:
+            return TAPPING_TERM + 70;
+        // ring
+        case HOME_S:
+        case HOME_L:
+            return TAPPING_TERM + 40;
+        // middle
+        case HOME_D:
+        case HOME_K:
+            return TAPPING_TERM + 20;
+        // index and thumb
+        default:
+            return TAPPING_TERM;
+    }
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```

Comment:
Seems that encoders are expected to be installed? This functionality should be moved to the keyboard level so that it works for QMK Configurator users. Same goes for the OLED code.

Reply:
Moved to a keyboard level with `*_user` calls support. Is it ok now?